### PR TITLE
os/bluestore: multithreaded '_kv_sync_thread'

### PR DESCRIPTION
**Pull request by tanghaodong25**
Signed-off-by: Haodong Tang <haodong.tang@intel.com>

### CODE REVIEW

**Comment by liewegas** [comment]
Did you observe a performance improvement with this?  I'm not sure that mulitple kv commit htreads is the way to go... it seems like it won't help too much given the way that rocksdb blocks it's commits (only one in flight to disk at a time, I think).

Also, I think we want to do something a bit different: ensure that each TransContext is submitted in its own thread, and the kv_sync_thread just loops and pushes them to stable storage.

**Comment by tanghaodong25** [comment]
@liewegas  There is performance improvement just when changing rocksdb to memdb. @majianpeng made more effort on optimizing multi-transaction-submission with a new PR #11189 . So I close it.

File: src/os/bluestore/BlueStore.h [hunk 1]
```diff
@@ -1144,11 +1145,16 @@ class BlueStore : public ObjectStore,
   vector<Finisher*> finishers;
 
   KVSyncThread kv_sync_thread;
+  vector<KVSyncThread *> kv_sync_threads;
   std::mutex kv_lock;
   std::condition_variable kv_cond, kv_sync_cond;
+  vector<std::mutex *> kv_locks;
```
Comment (yuyuyu101) [inline src/os/bluestore/BlueStore.h]: vector<std::mutex> is ok

File: src/os/bluestore/BlueStore.h [hunk 2]
```diff
@@ -1144,11 +1145,16 @@ class BlueStore : public ObjectStore,
...
+  vector<std::condition_variable *> kv_conds;
```
Comment (yuyuyu101) [inline src/os/bluestore/BlueStore.h]: same

**Pull request closed without merging**